### ISSUE TITLE
FORCEWEB- CTD

### ISSUE THREAD

**Issue by witchiewoman** [OpenApoc/OpenApoc#1103]
After researching "Forceweb" and producing this item anytime you hover over this item to equip in inventory it immediately causes the game to CTD. Also navigating to the info page of Forceweb in the dictionary screen causes a CTD.

I have no mods enabled and tool tips set to off just in case

intel I7 Nvidia 2070
32GB ram on windows 10/11

**Comment by witchiewoman** [comment]
[openapoc_log.txt](https://github.com/OpenApoc/OpenApoc/files/9603743/openapoc_log.txt)

**Comment by witchiewoman** [comment]
additional files here
[log.txt](https://github.com/OpenApoc/OpenApoc/files/9603770/log.txt)
[settings.conf.txt](https://github.com/OpenApoc/OpenApoc/files/9603771/settings.conf.txt)

**Comment by FilmBoy84** [comment]
Many thanks for the report @witchiewoman 
I notice you are on the Extended Weapons Mod
Can you please confirm which version you are using?
The ForceWeb is a cut weapon from the Original Game that is labelled WIP in current versions due to being incomplete
But i know some older versions of the EWM will not inform you at the research stage

**Comment by witchiewoman** [comment]
I am using  most current on site, is there anyway to edit my save game 
to remove forceweb in my manufactured goods and dictionary so it doesnt 
cause crashes?

**Comment by FilmBoy84** [comment]
No problem, if you can provide me the save I can walk you through fixing it (it's save dependant as I need to get the ID for the project from the save archive)

Nonetheless, it's usually, under the unpacked save file, in `research.xm`l and look for the `<current_project>` that refers to psi-grenade production - you can then usually delete this line and it will stop it being produced :) Then repack the save and load the game

![image](https://user-images.githubusercontent.com/33913088/192114679-b4d67b3d-fd0b-4c3a-9e33-27bbd413740a.png)

For future reference, when researching things in the latest EWM (v9 Preview) versions, all research projects labelled WIP may or may not crash the game

Nonetheless, by reporting this error with the ForceWeb, you've set me about fixing it - so thanks for the report 😄 
I'll update the Git repot with a fix in the next EWM update

![image](https://user-images.githubusercontent.com/33913088/192114268-65e53acc-3a1c-45a3-9b34-9b6d51c8a17c.png)

**Comment by FilmBoy84** [comment]
Actually, it's better to delete the entire set of lines as follows
![image](https://user-images.githubusercontent.com/33913088/192114887-f1901705-b232-4220-8ff2-f72e580b167f.png)
This will then remove the project and the agents assigned to avoid any NoAgentID errors :)

**Comment by witchiewoman** [comment]
weird there is no saves folder in openpoc but i do have an ongoing saved 
game. The forceweb was already reasearched and i have 2 manufactured in 
inventory. But i dont dare hovr over them to equip as it CTD's 
immediately, same with entry in dictionary. Is save located somewhere 
else?

**Comment by FilmBoy84** [comment]
Unless you have a file called portable.txt where the openapoc.exe is located (or selected portable in the installer) all saves should be somewhere under `%APPDATA%` then something like `OpenApoc/Saves`

Sorry, i have mine set up as portable already so cannot screenshot a location as an example

You can make OpenApoc portable by adding a portable.txt to the executable location then copying all the stuff from Appdata to there too (retaining structure where applicable)
[portable.txt](https://github.com/OpenApoc/OpenApoc/files/9639497/portable.txt)

**Comment by witchiewoman** [comment]
found and attached it

**Comment by FilmBoy84** [comment]
Sadly i did not get attachment via email reply, can you add to the git issue directly when you have chance please?

![image](https://user-images.githubusercontent.com/33913088/192117917-74109f84-6eec-4f6d-9f97-56a28886fb7e.png)

**Comment by witchiewoman** [comment]
ok uploading but removing psi-grenade entry did not remove forceweb, 
instead it knocked out incendary grenade and my motion scanners.

**Comment by witchiewoman** [comment]
save attached
[save_m2 2022.txt](https://github.com/OpenApoc/OpenApoc/files/9639541/save_m2.2022.txt)

**Comment by FilmBoy84** [comment]
Have you got a save before you edited?

I cannot find any reference to forceweb being researched in the one you sent?

My apologies, my reply about what to delete was not particularly well worded - I was using psi grenade as an example as I had a game to hand where this, not the forceweb, was in production - you needed to delete the equivalent lab entry for forceweb (not the research item) under a `<current_project>` tab

If you send me the broken save, before you made any edits I'll fix it :)

**Comment by witchiewoman** [comment]
Sadly I do not but I did what you said and forceweb it still in 
completed research along with 3 manufactured in economy inventory. But 
now incendiary grenades and motion scanners are greyed out and cannot be 
purchased. You instructed me to remove sci grenade, thats apparently not 
"Forceweb"

**Comment by FilmBoy84** [comment]
[save_m2.zip](https://github.com/OpenApoc/OpenApoc/files/9639589/save_m2.zip)
Please find attached zip with the save in - you should be able to pull out the `save_m2.save` file and put it to your saves directory, then load the game and load the save

I have fixed the economy.xml to have the items you listed as missing back in stock - i'm not sure why they disappeared - they are unrelated to research

Possibly another instance of #1070 at play where the marketplace gains/loses stuff erratically at some point each week

![image](https://user-images.githubusercontent.com/33913088/192118912-7d2f6155-0e63-4fa2-97d5-15222fc72af4.png)
![image](https://user-images.githubusercontent.com/33913088/192118914-99a6502c-2280-429a-b98a-5c20fab2ff66.png)

Let me know if anything else is amiss

The beauty of OpenApoc is that 90% of the time, if there's an error like this, it's solution is somewhere in the XML

**Comment by witchiewoman** [comment]
On 2022-09-24 2:15 pm, Jacob Deuchar wrote:
Thanks will try

**Comment by witchiewoman** [comment]
On 2022-09-24 2:15 pm, Jacob Deuchar wrote:
Hi ok that fixed my save but forceweb is still in manufactured items (3 
of them) if I mouse over them th game does CTD. Is there any way to 
delete the 3 in inventory and remove the entry in the encyclopedia? 
These 2 things are causing the crash.

**Comment by FilmBoy84** [comment]
Yep, I'll use the save i gave back to you above to edit a new one

First thing i do after unpacking the save is to look for `player_bases.xml` and search for the forceweb in the list

I soon find the three of them in the list like this

![image](https://user-images.githubusercontent.com/33913088/192119550-26541729-10e6-4fc9-a736-aabfcb6f0daa.png)

I select the following (grey highlight)
![image](https://user-images.githubusercontent.com/33913088/192119569-1171a007-8c43-4f08-8c36-f4eebaf9fc2d.png)

And delete

It now looks like this

![image](https://user-images.githubusercontent.com/33913088/192119582-8c983a02-a7e3-4d9e-99fb-f3a2c36536d3.png)

I can then repack the savegame (see attached fixed version below)

and they should now be gone when loaded

![image](https://user-images.githubusercontent.com/33913088/192119640-46245acc-8c3a-4c83-b7a3-a2c68d2b2ba8.png)
[save_m2.zip](https://github.com/OpenApoc/OpenApoc/files/9639632/save_m2.zip)

**Comment by FilmBoy84** [comment]
I can't remove the forceweb from UFOPaedia in this version of the EWM sadly

But as long as i don't click it, i can use the rest of the paedia it appears :)

![image](https://user-images.githubusercontent.com/33913088/192119942-6d4db23a-14d5-44ed-956d-bbe94c6072c8.png)
![image](https://user-images.githubusercontent.com/33913088/192119950-444fca9d-0f08-4b6a-aaf7-9e0320d9ffd5.png)
![image](https://user-images.githubusercontent.com/33913088/192119958-35cbcf85-3bbe-4018-806d-f90a250c3237.png)

**Comment by witchiewoman** [comment]
I am glad I learned not to reaearch WIP items out of this. :)

**Comment by FilmBoy84** [comment]
Apologies for all the trouble

Let me know if the game is now working as expected and I'll close the issue

**Comment by witchiewoman** [comment]
Back to normal, thanks a bunch!

**Comment by FilmBoy84** [comment]
Fantastic, thanks for letting me know
Enjoy!
